src/PluginChain.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Common;
+
+use Http\Client\Common\Exception\LoopException;
+use Http\Promise\Promise;
+use Psr\Http\Message\RequestInterface;
+
+final class PluginChain
+{
+    /** @var Plugin[] */
+    private $plugins;
+
+    /** @var callable */
+    private $clientCallable;
+
+    /** @var int */
+    private $maxRestarts;
+
+    /** @var int */
+    private $restarts = 0;
+
+    /**
+     * @param Plugin[] $plugins
+     */
+    public function __construct(array $plugins, callable $clientCallable, int $maxRestarts)
```

Comment:
Lets do `public function __construct(array $plugins, callable $clientCallable, array $options)` instead. That will allow us to add more options without breaking BC

Reply:
Done!